src/io.rs

```diff
@@ -89,27 +117,83 @@ pub(crate) fn build_output(
     // the desired number of frames.
     buffer_size = (buffer_size + BUFFER_SIZE - 1) / BUFFER_SIZE * BUFFER_SIZE;
 
-    let mut config: StreamConfig = supported_config.into();
-    config.buffer_size = cpal::BufferSize::Fixed(buffer_size);
+    let mut primary_config: StreamConfig = supported_config.clone().into();
+    primary_config.buffer_size = cpal::BufferSize::Fixed(buffer_size);
+
+    let mut config: StreamConfig = supported_config.clone().into();
+
+    match options {
+        Some(opts) => {
+            config.sample_rate = CpalSampleRate(opts.sample_rate.unwrap_or(config.sample_rate.0));
+            config.channels = opts.channels.unwrap_or(config.channels);
+
+            match opts.latency_hint {
```

Comment:
Can you move this block into a helper function, makes it a bit more readable and then we can re-use if for the InputBuilder